src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class FileWork {
+    private static final String SPECIFIED_CHARACTER = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        List<String> arrayList = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file));) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(" ");
+                value = bufferedReader.readLine();
+            }
+            String[] allWords = stringBuilder.toString().toLowerCase().split(" ");
+            for (String word: allWords) {
+                if (word.startsWith(SPECIFIED_CHARACTER)) {
+                    arrayList.add(word.replaceAll("[.?!]", ""));
+                }
```

Comment:
```suggestion
            String[] allWords = stringBuilder.toString().toLowerCase().split("\\W+");
            for (String word: allWords) {
                if (word.startsWith(SPECIFIED_CHARACTER)) {
                    arrayList.add(word);
                }
```
Find out what this regex does and test if it works

Reply:
thanks 
i forgot this regex